core/signal_handler.cpp

```diff
@@ -135,9 +135,25 @@ namespace MR
 
     void mark_file_for_deletion (const std::string& filename)
```

Comment:
Upstream of this, there has already been added in #2125 a `static bool` that controls whether or not piped images are deleted. It would IMO make more sense to give the ability to modulate the value of that boolean based on an environment variable, rather than having two sequential if statements in two different locations in the code controlling whether or not any given image is marked for deletion.

Reply:
Addressed in [2894b1d](https://github.com/MRtrix3/mrtrix3/pull/2678/commits/2894b1dc392dd56aa068b359544ce199cde0422c).